Userland/Utilities/mktemp.cpp

```diff
@@ -7,66 +7,60 @@
 #include <AK/LexicalPath.h>
 #include <AK/Random.h>
 #include <LibCore/ArgsParser.h>
-#include <errno.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
 #include <fcntl.h>
-#include <stdio.h>
-#include <string.h>
 #include <sys/stat.h>
 #include <unistd.h>
 
-constexpr const char* default_template = "tmp.XXXXXXXXXX";
+constexpr StringView default_template = "tmp.XXXXXXXXXX";
 
-static char* generate_random_filename(const char* pattern)
+static ErrorOr<String> generate_random_filename(String const& pattern)
 {
-    char* new_filename = strdup(pattern);
-    int pattern_length = strlen(pattern);
+    auto new_filename = StringBuilder { pattern.length() };
 
     constexpr char random_characters[] = "abcdefghijklmnopqrstuvwxyz0123456789";
-    for (auto i = pattern_length - 1; i >= 0; --i) {
-        if (pattern[i] != 'X')
-            break;
-        new_filename[i] = random_characters[(get_random<u32>() % (sizeof(random_characters) - 1))];
+    for (size_t i = 0; i < pattern.length(); i++) {
+        if (pattern[i] == 'X')
+            TRY(new_filename.try_append(random_characters[(get_random<u32>() % (sizeof(random_characters) - 1))]));
+        else
+            TRY(new_filename.try_append(pattern[i]));
```

Comment:
These _cannot_ fail.
the allocation above in the `StringBuilder` ctor however, can.